__Задача 1__

Известно, что генеральная совокупность распределена нормально с известным средним квадратическим отклонением 16. Найти ширину доверительного интервала для оценки математического ожидания `mu` с надёжностью 0.95, если выборочное среднее равно 80, а объём выборки равен 256.

In [8]:
import numpy as np
from scipy import stats

In [3]:
s_mean = 80
n = 256
alpha = 0.05

In [7]:
z = (s_mean - 1) / (alpha / np.sqrt(n))
z

25280.0

In [13]:
t1 = stats.norm.ppf(alpha / 2)
t2 = stats.norm.ppf(1 - alpha / 2)
round(t2 - t1, 8)

3.91992797

Ответ: 3.91992797

__Задача 2__

Продавец утверждает, что средний вес пачки печенья составляет 200 г. Из партии извлечена выборка из 10 пачек. Вес каждой пачки составляет:
```
202, 203, 199, 197, 195, 201, 200, 204, 194, 190
```
Известно, что их веса распределены нормально. Верно ли утверждение продавца, если учитывать, что уровень значимости равен 1%? Найдите P-значение для данного теста и укажите его в качестве ответа.

In [30]:
cookies = np.array([202, 203, 199, 197, 195, 201, 200, 204, 194, 190])

In [66]:
def statistic(samples: np.ndarray) -> float:
    return (samples.mean() - mu) / (samples.std(ddof=1) / np.sqrt(samples.shape[0]))

In [67]:
alpha = 0.01
mu = 200
n = cookies.shape[0]
S = statistic(cookies)
t1 = stats.t.ppf(alpha / 2, df=n - 1)
t2 = stats.t.ppf(1 - alpha / 2, df=n - 1)

In [68]:
p_left = stats.t.cdf(S, df=n - 1)
p_right = 1 - stats.t.cdf(S, df=n - 1)
pvalue = 2 * min(p_left, p_right)

In [69]:
S, t1, t2, pvalue

(-1.0651074037450896,
 -3.24983554401537,
 3.2498355440153697,
 0.31456660352472854)

In [70]:
stats.ttest_1samp(cookies, popmean=200)

Ttest_1sampResult(statistic=-1.0651074037450896, pvalue=0.31456660352472854)

__Ответ__: 0.3145666

__Задача 3__

Дана выборка диаметров подшипников из примера 1 с занятия 5:
```
samples = [0.6603, 0.9466, 0.5968, 1.3792, 1.5481, 0.7515, 1.0681, 1.1134,
           1.2088, 1.701 , 1.0282, 1.3579, 1.0191, 1.1784, 1.1168, 1.1372,
           0.7273, 1.3958, 0.8665, 1.5112, 1.161 , 1.0232, 1.0865, 1.02  ]
```
Предполагая, что диаметры подшипников распределены нормально, проверьте гипотезу о том, что дисперсия случайной величины равна `0.0625` при уровне значимости `alpha = 0.05`. Что для этого нужно знать:
1. Альтернативная гипотеза двухсторонняя.
2. Статистика для теста: `H = (n - 1) * sample_variance / variance`, где `n` - число элементов в выборке, `sample_variance` - несмещённая оценка дисперсии, `variance` - утверждаемая нулевой гипотезой дисперсия.
3. Эта статистика в предположении верности нулевой гипотезы имеет распределение хи-квадрат с параметром `df = n - 1`. Её квантили можно найти с помощью функции `scipy.stats.chi2.ppf`.

В качестве ответа запишите 1, если нулевая гипотеза верна, и 0 иначе.



In [42]:
samples = np.array([0.6603, 0.9466, 0.5968, 1.3792, 1.5481, 0.7515, 1.0681, 1.1134,
           1.2088, 1.701 , 1.0282, 1.3579, 1.0191, 1.1784, 1.1168, 1.1372,
           0.7273, 1.3958, 0.8665, 1.5112, 1.161 , 1.0232, 1.0865, 1.02  ])

In [62]:
alpha = 0.05
n = samples.shape[0]
sample_variance = samples.var(ddof=1)
variance = 0.0625

In [63]:
H = (n - 1) * sample_variance / variance
H

28.720541753333343

In [64]:
t1 = stats.chi2.ppf(alpha / 2, df = n - 1)
t2 = stats.chi2.ppf(1 - alpha / 2, df = n - 1)

t1, t2

(11.688551922452438, 38.0756272503558)

__Ответ__: 1